Ghidra/Features/Base/src/main/java/ghidra/app/util/exporter/BinExport2Builder.java

```diff
@@ -0,0 +1,501 @@
+ *      implementation</a>
+ */
+public class BinExport2Builder {
+	private final Builder builder = BinExport2.newBuilder();
+
+	private TaskMonitor monitor;
+
+	private final Program program;
+	private final Listing listing;
+	private final BasicBlockModel bbModel;
+
+	private BinExportExporter.MnemonicMapper mnemonicMapper =
+		new BinExportExporter.IdentityMnemonicMapper();
+	private long addressOffset = 0;
+	private boolean prependNamespace = false;
+
+	public BinExport2Builder(Program ghidraProgram) {
+		program = ghidraProgram;
+		listing = program.getListing();
+		bbModel = new BasicBlockModel(program, true);
+	}
+
+	public BinExport2Builder setMnemonicMapper(BinExportExporter.MnemonicMapper mapper) {
+		mnemonicMapper = mapper;
+		return this;
+	}
+
+	public BinExport2Builder setAddressOffset(long offset) {
+		addressOffset = offset;
+		return this;
+	}
+
+	public BinExport2Builder setPrependNamespace(boolean isPrepended) {
+		prependNamespace = isPrepended;
+		return this;
+	}
+
+	private long getMappedAddress(Address address) {
+		return address.getOffset() - addressOffset;
+	}
+
+	private long getMappedAddress(Instruction instr) {
+		return getMappedAddress(instr.getAddress());
+	}
+
+	private void buildMetaInformation() {
+		monitor.setIndeterminate(true);
+		monitor.setMessage("Exporting meta data");
+
+		// Ghidra uses a quad format like x86:LE:32:default, BinExport just keeps
+		// the processor and address size.
+		String[] quad = program.getLanguageID().toString().split(":", 4);
+		// TODO(cblichmann): Canonicalize architecture names
+		String arch = quad[0] + "-" + quad[2];
+
+		builder.getMetaInformationBuilder()
+				.setExecutableName(new File(program.getExecutablePath()).getName())
+				.setExecutableId(program.getExecutableSHA256())
+				.setArchitectureName(arch)
+				.setTimestamp(System.currentTimeMillis() / 1000);
+	}
+
+	private void buildExpressions(Map<String, Integer> expressionIndices) {
+		var cuf = new CodeUnitFormat(new CodeUnitFormatOptions());
+		int id = 0;
+		for (Instruction instr : listing.getInstructions(true)) {
+			for (int i = 0; i < instr.getNumOperands(); i++) {
+				String opRep = cuf.getOperandRepresentationString(instr, i);
+				if (expressionIndices.putIfAbsent(opRep, id) != null) {
+					continue;
+				}
+				id++;
+				builder.addExpressionBuilder()
+						.setType(BinExport2.Expression.Type.SYMBOL)
+						.setSymbol(opRep);
+			}
+		}
+	}
+
+	private void buildOperands(Map<String, Integer> expressionIndices) {
+		var entries = new ArrayList<>(expressionIndices.entrySet());
+		entries.sort(Entry.comparingByValue());
+		for (var entry : entries) {
+			builder.addOperandBuilder().addExpressionIndex(entry.getValue());
+		}
+	}
+
+	private void buildMnemonics(Map<String, Integer> mnemonicIndices) {
+		monitor.setIndeterminate(true);
+		monitor.setMessage("Computing mnemonic histogram");
+		var mnemonicHist = new HashMap<String, Integer>();
```

Comment:
Our convention is to not use `var` in our code.

Reply:
Done. I hope I didn't break the PR by updating and rebasing different workstations...